themes/jquery/functions.php

```diff
@@ -251,3 +251,39 @@ function jq_image_posted_on() {
 
 	return $classes;
 } );
+
+/**
+ * Content Security Policy
+ */
+function jq_content_security_policy() {
+	$nonce = wp_create_nonce( JQUERY_LIVE_SITE );
```

Comment:
I believe these nonces mean something else than for CSP. These WordPress nonces are CSRF tokens, each "create" call stores this token in a database, to give permission for the current user to perform a certain action once, and then in a subsequent request you can call wp_verify_nonce to verify that action.

Thus calling it would have the downside of storing unused data in the database. If all we want is a random string that is different on each page, and different each time the same page is regenerated after it expires from cache, then something like random_bytes would do the same:

* https://www.php.net/random_bytes

```php
$nonce = bin2hex( random_bytes( 8 ) );
```

However, I don't see this `$once` used anywhere. Did it miss it? Afaik one woudl set this in the CSP header in order to allow `<script once="…">` in a way that allows only those inline scripts and nothing else.

Reply:
I haven't used it anywhere yet. I was trying to make it work for API demos, but it turned out to be too complicated with the HTML string coming from the database. I left the first step here in case we ever need to use nonce, or in case I missed a place that we'll discover later from the reports on stage.

Reply:
Now using `random_bytes`